util/keygen.go

```diff
@@ -0,0 +1,43 @@
+package util
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// KeyPair simply holds a private-public key pair as strings, with no extra information.
+type KeyPair struct {
+	Private string
+	Public  string
+}
+
+// MakeKeyPair generates and returns a private-public key pair.
+func MakeKeyPair() (*KeyPair, error) {
+	privkey, err := rsa.GenerateKey(rand.Reader, 2048) // TODO: Key size as parameter
```

Comment:
Consider using Ed25519 keys, because they are [better](https://blog.g3rt.nl/upgrade-your-ssh-keys.html). Supported by the [ed25519 package](https://godoc.org/golang.org/x/crypto/ed25519). NB: They are not supported by gnome-keyring (and its agent) though, but I guess this code doesn't make use if it anyway, right?